### ISSUE TITLE
Signing a manual transaction on an unsync'ed node generates an invalid signature.

### ISSUE THREAD

**Issue by greatwolf** [BTCGPU/BTCGPU#419]
<-- Describe the issue -->
 When manually crafting and signing a transaction using the debug console, if done on a new node with no blockchain downloaded, it generates an invalid signature. The network will reject the broadcasted transaction with `-26: mandatory-script-verify-flag-failed (Signature must be zero for failed CHECK(MULTI)SIG operation)`

<--- What behavior did you expect? -->
A fully signed transaction that the network considers valid and acceptable when broadcasted.

<--- What was the actual behavior (provide screenshots if the issue is GUI-related)? -->
Broadcasting the manually created signed transaction gets rejected by the network with the error:

    -26: mandatory-script-verify-flag-failed (Signature must be zero for failed CHECK(MULTI)SIG operation)

<--- How reliably can you reproduce the issue, what are the steps to do so? -->
This can be reproduced 100%. Start by downloading the node from release section. Extract and run ./bin/bitcoin-qt.exe
Open the debug console, manually craft a transaction from a valid utxo using `createrawtransaction` and `signrawtransactionwithkey` or `signrawtransactionwithwallet`. Broadcast the final transaction hex output.

<-- What version of Bitcoin Core are you using, where did you get it (website, self-compiled, etc)? -->
This is with the latest version of Bitcoin Gold reference client 0.17.3 from the release section. https://github.com/BTCGPU/BTCGPU/releases

<-- What type of machine are you observing the error on (OS/CPU and disk type)? -->
This is on Windows 7 SP1 64-bit. Also tested it on VM VirtualBox Linux Mint distribution.

<-- Any extra information that might be useful in the debugging process. -->
I explained this in more detail in the bitcoin gold discord dev channel: https://discord.com/channels/378266958824603649/399179868568289280/838159738948681739

TL;DR; it appears the problem is that the `prevtx.amount` field given for `signrawtransaction*` methods are ignored in an unsync'ed node. That field is needed for bip143 replay protection employed during the fork. In the case of an unsync node, `signrawtransaction*` always generates the same hex output for the transaction regardless of what's passed in as `prevtx.amount` even if it's completely bogus.

**Comment by MentalNomad** [comment]
"That field is needed for bip143 replay protection employed during the fork."

I wonder if a partially-synced node will stop exhibiting this behavior after it passes height 491407 (the fork.) At that point, the validation logic changes.

You mentioned on Discord that the error did not occur on the testnet node: the BTG testnet was reset, relative to Bitcoin, and has a much shorter block height before the BTG fork rules kick in; it may be hard or impossible to catch a testnet node in a pre-fork condition.

**Comment by bitcoingold-readonly** [comment]
I’m pretty sure that’s the case. Testnet has the Hardfork activation block
height at 1. So you cannot really observe the behavior.
